apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/Session.cs

```diff
@@ -0,0 +1,175 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Google.Apis.Auth.OAuth2;
+
+namespace Google.Cloud.Spanner.V1
+{
+    public partial class Session : IDisposable
+    {
+        private readonly CancellationTokenSource _cancellation = new CancellationTokenSource();
+        private Task _keepAlive;
+        private int _lastAccessTime = Environment.TickCount;
+        private ITokenAccess _credential;
+        private static int _activeSessions;
+        private static readonly object CreateSync = new object();
+
+        internal void InitializePoolManagedSession(ITokenAccess credential)
+        {
+            _credential = credential;
+            _keepAlive = Task.Run(KeepAlive, _cancellation.Token);
+        }
+
+        internal ITokenAccess AssociatedCredential => _credential;
+
+        private async Task KeepAlive()
+        {
+            ExecuteSqlRequest request = new ExecuteSqlRequest
+            {
+                SessionAsSessionName = SessionName,
+                Sql = "SELECT 1",
+            };
+
+            while (true)
```

Comment:
Will properly review this logic later, particularly if the types are going to be changed. Note that if we keep using `TickCount`, we need to carefully handle overflow.